RocketChat/Rocket.Chat
# Regression: 0 appearing on sidebar on specific occasions

## Proposed changes (including videos or screenshots)


Fixed faulty logic


## Issue(s)


## Steps to test or reproduce

- Create a room
- Go to notification preferences in that room
- Disable "Mark as Unread"
- Enable "Show badge for mentions"

A 0 will appear when there's no unread messages.

## Further comments

TC-475
